client/src/store/reducers/todoReducer.ts

```diff
@@ -51,13 +55,18 @@ export const todoSlice = createSlice({
     builder.addCase(fetchAddTask.fulfilled, (state, action) => {
       state.todos.unshift(action.payload);
       state.allTodosExist = true;
-      state.query.search = "";
+      state.query = { search: "", filter: FilterValue.ALL };
     });
     builder.addCase(fetchEditTask.fulfilled, (state, action) => {
-      const changedTask = action.payload;
-      state.todos = state.todos.map((task) =>
-        task._id === changedTask._id ? changedTask : task
-      );
+      const updatedTask = action.payload;
+      state.todos = state.todos
+        .map((task) => (task._id === updatedTask._id ? updatedTask : task))
+        .filter(
+          (task) =>
+            (task.completed && state.query.filter === FilterValue.COMPLETED) ||
+            (!task.completed && state.query.filter === FilterValue.ACTIVE) ||
+            state.query.filter === FilterValue.ALL
```

Comment:
You can use destructuring for state.query.filter to make it shorter as you use it 3 times

Reply:
Fixed -  "const { filter } = state.query;"